### PR TITLE
riak-cs: add to nixpkgs

### PR DESCRIPTION
**Pull request by mdaiter**
###### Motivation for this change
Needed to add in specific Riak CS bindings when dealing with the [Riak CS](http://docs.basho.com/riak/cs/2.1.1/) package from Basho. Coordinated with this merge request: https://github.com/NixOS/nixpkgs/pull/20128 .

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] OS X
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Review by mdaiter — COMMENTED** [review]
riak-cs: assigned 261 instead of 260 because of the stanchion package submission

**Comment by mdaiter** [comment]
cc @cstrahan

File: nixos/modules/services/databases/riak-cs.nix [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+  options = {
+
+    services.riak-cs = {
+
+      enable = mkEnableOption "riak-cs";
+
+      package = mkOption {
+        type = types.package;
+        example = literalExample "pkgs.riak2";
```
Comment (LnL7) [inline nixos/modules/services/databases/riak-cs.nix]: we don't have that attribute anymore

**Review by LnL7 — COMMENTED** [review]
there are a couple of comments on the stanchion pr that also apply here

File: nixos/modules/services/databases/riak-cs.nix [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+      package = mkOption {
+        type = types.package;
+        example = literalExample "pkgs.riak";
+        description = ''
+          Riak package to use.
+        '';
+      };
+
+      nodeName = mkOption {
+        type = types.string;
```
Comment (ericsagnes) [inline nixos/modules/services/databases/riak-cs.nix]: For this and all following `types.string` usages, it is recommended to use `types.str` instead. ([reason](https://github.com/NixOS/nixpkgs/blob/master/lib/types.nix#L141-L143))

Comment (mdaiter) [inline nixos/modules/services/databases/riak-cs.nix]: I'll change it

Comment (mdaiter) [inline nixos/modules/services/databases/riak-cs.nix]: Done

**Comment by LnL7** [comment]
I run into some issues with this, I think the build is impure.

```
fatal: Not a git repository (or any of the parent directories): .git
--2016-11-18 18:40:18--  https://raw.githubusercontent.com/basho/bitcask/develop/priv/scripts/downgrade_bitcask.erl
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... failed: Name or service not known.
wget: unable to resolve host address 'raw.githubusercontent.com'
make: *** [Makefile:29: riak_test/src/downgrade_bitcask.erl] Error 4
builder for ‘/nix/store/5ls6m2p7vv2m3hw8sy5shqysdxabqpiv-riak_cs-2.1.1.drv’ failed with exit code 2
error: build of ‘/nix/store/5ls6m2p7vv2m3hw8sy5shqysdxabqpiv-riak_cs-2.1.1.drv’ failed
```

**Review by LnL7 — APPROVED** [review]

File: nixos/modules/services/databases/riak-cs.nix [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+  ###### interface
+
+  options = {
+
+    services.riak-cs = {
+
+      enable = mkEnableOption "riak-cs";
+
+      package = mkOption {
+        type = types.package;
```
Comment (mdaiter) [inline nixos/modules/services/databases/riak-cs.nix]: Done

File: nixos/modules/services/databases/riak-cs.nix [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+        default = "pkgs.riak-cs";
```
Comment (LnL7) [inline nixos/modules/services/databases/riak-cs.nix]: you need the actual package here, not a string.

Comment (mdaiter) [inline nixos/modules/services/databases/riak-cs.nix]: done

**Comment by falsifian** [comment]
@LnL7 it looks like Travis is failing pull requests because of something to do with riak_cs. [Here's an example](https://travis-ci.org/NixOS/nixpkgs/jobs/177536863) from #20593.

The particular error is `Package ‘riak_cs-2.1.1’ in ‘/home/travis/build/NixOS/nixpkgs/pkgs/servers/nosql/riak-cs/2.1.1.nix:62’ is not supported on ‘i686-linux’, refusing to evaluate.`

Sorry to make more work for you, but could you try to figure out why this is happening?

I'm going to go ahead and revert this since it's making Travis angry; please push again once you've got it working. If I'm mistaken about this somehow, then my apologies; please revert my revert in that case :-)

I suppose you can reproduce the build error by running the command in the Travis log on an i686 machine: `./maintainers/scripts/travis-nox-review-pr.sh nixpkgs-verify nixpkgs-manual nixpkgs-tarball`

**Comment by falsifian** [comment]
@mdaiter

**Pull request merged**